contracts/protocol/lendingpool/Looping.sol

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.6.12;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ILendingPool} from '../../interfaces/ILendingPool.sol';
+import {ILendingPoolAddressesProvider} from '../../interfaces/ILendingPoolAddressesProvider.sol';
+import {IFlashLoanReceiver} from "../../flashloan/interfaces/IFlashLoanReceiver.sol";
+import "../../misc/interfaces/IWETH.sol";
+
+contract Looping is IFlashLoanReceiver {
+  using SafeERC20 for IERC20;
+  using SafeMath for uint256;
+
+  ILendingPoolAddressesProvider public immutable override ADDRESSES_PROVIDER;
+  ILendingPool public immutable override LENDING_POOL;
+  IWETH internal immutable WETH;
+
+  constructor(ILendingPoolAddressesProvider provider, IWETH weth) public {
+    ADDRESSES_PROVIDER = provider;
+    LENDING_POOL = ILendingPool(provider.getLendingPool());
+    WETH = weth;
+  }
+
+  function getWETHAddress() external view returns (address) {
+    return address(WETH);
+  }
+
+  /// This function is called by lending pool after your contract has received the flash loaned amount
+  /// @param assets  flashloan received assets
+  /// @param amounts  flashloan received asset amounts
+  /// @param premiums  flashloan assets premium amounts
+  /// @param initiator  flashloan initiator
+  /// @param params  flashloan parameters
+  function executeOperation(
+    address[] calldata assets,
+    uint256[] calldata amounts,
+    uint256[] calldata premiums,
+    address initiator,
```

Comment:
verify that `initiator` is this contract otherwise other contract can initiate flashload again this